#!/usr/bin/env python
PACKAGE = "guidance"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("enable_imu",               bool_t,   0, "Enable IMU datastream",  True)
gen.add("enable_utrasonic",         bool_t,   0, "Enable  datastream",  False)
gen.add("enable_obstacle_distance", bool_t,   0, "Enable  datastream",  True)
gen.add("enable_velocity",          bool_t,   0, "Enable  datastream",  False)
gen.add("enable_motion",            bool_t,   0, "Enable  datastream",  True)
gen.add("enable_depth_front",       bool_t,   0, "Enable depth image for front",    True)
gen.add("enable_depth_right",       bool_t,   0, "Enable depth image for right",    False)
gen.add("enable_depth_back",        bool_t,   0, "Enable depth image for back",     False)
gen.add("enable_depth_left",        bool_t,   0, "Enable depth image for left",     False)
gen.add("enable_depth_bottom",      bool_t,   0, "Enable depth image for bottom",   False)
gen.add("enable_disparity_front",       bool_t,   0, "Enable disparity image for front",    True)
gen.add("enable_disparity_right",       bool_t,   0, "Enable disparity image for right",    False)
gen.add("enable_disparity_back",        bool_t,   0, "Enable disparity image for back",     False)
gen.add("enable_disparity_left",        bool_t,   0, "Enable disparity image for left",     False)
gen.add("enable_disparity_bottom",      bool_t,   0, "Enable disparity image for bottom",   False)
gen.add("enable_camera_front_left",      bool_t,   0, "Enable raw stereo image from front left",  True)
gen.add("enable_camera_right_left",      bool_t,   0, "Enable raw stereo image from right left",  False)
gen.add("enable_camera_back_left",       bool_t,   0, "Enable raw stereo image from back left",   False)
gen.add("enable_camera_left_left",       bool_t,   0, "Enable raw stereo image from left left",   False)
gen.add("enable_camera_bottom_left",     bool_t,   0, "Enable raw stereo image from bottom left", False)
gen.add("enable_camera_front_right",      bool_t,   0, "Enable raw stereo image from front right",  True)
gen.add("enable_camera_right_right",      bool_t,   0, "Enable raw stereo image from right right",  False)
gen.add("enable_camera_back_right",       bool_t,   0, "Enable raw stereo image from back right",   False)
gen.add("enable_camera_left_right",       bool_t,   0, "Enable raw stereo image from left right",   False)
gen.add("enable_camera_bottom_right",     bool_t,   0, "Enable raw stereo image from bottom right", False)

gen.add("filterSpeckles", bool_t, 0, "Filter out speckles", True)
gen.add("maxSpeckleSize", int_t, 0, "maxSpeckleSize", 120, 0, 400)
gen.add("maxDiff", double_t, 0, "maxSpeckleSize", 20, 0, 400)

exit(gen.generate(PACKAGE, "guidance_node", "guidance"))
